drivers/can/can_mcp251xfd.c

```diff
@@ -1622,48 +1624,57 @@ static int mcp251xfd_init(const struct device *dev)
 
 	ret = mcp251xfd_init_con_reg(dev);
 	if (ret < 0) {
+		ret = -EIO;
```

Comment:
I think it will be cleaner just to forward the error at these points and get rid of the done label.